apps/functions/src/app/player.ts

```diff
@@ -1,12 +1,33 @@
-import { utils } from 'ethers';
+import { utils, ethers } from 'ethers';
 import { SignedMessage } from '@nft/model';
+import env from '@nft/env';
+import * as abi from '@nft/model/erc1155.json';
+import * as request from "request-promise";
 
 export const checkSignature = async (data: SignedMessage): Promise<string> => {
   const { message, signature, tokenId } = data;
-
   const ethAddress = utils.verifyMessage(message, signature);
   console.log(`Ethereum address : ${ethAddress}`, `Token Id: ${tokenId}`);
-  //@TODO check ownership against ETH contract
 
-  return 'https://foo.bar.com';
+  const provider = ethers.getDefaultProvider(env.eth.network);
+  const contract = new ethers.Contract(env.eth.erc1155, abi, provider);
+
+  const balance = await contract.balanceOf(ethAddress, tokenId);
+
+  if (balance.toNumber() > 0) {
+    const titleId = utils.formatEther(balance);
```

Comment:
I think this line is wrong, `titleId` is `tokenId`, it's the same,
`balance` will be how much dvd of this movie do I own

Reply:
ok .updated